### PR TITLE
PLAT-911 | Content classifier on content side

### PR DESCRIPTION
**Pull request by rogatty**
https://bethink.atlassian.net/browse/PLAT-911

TODO:
- [x] Move shared stores and components from `admin` directory
- [x] Update terms state in annotations editor
- [x] Update terms state in quiz questions
- [x] Update terms state in flashcards
- [x] Add content item classifier to slides
- [x] Fix the freeze when the term autocomplete is used for the first time
- [x] Expand content item classifier in admin
- [x] Hide the counter in a single item context
- [x] Fix terms includes after #1256 is merged

### CODE REVIEW

File: resources/assets/js/admin/components/contentClassifier/ContentItemClassifierEditor.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
+<template>
+	<div v-if="canAccess" class="content-item-classifier">
+		<div v-if="expanded" class="content-item-classifier__editor">
```
Comment (bognix) [inline resources/assets/js/admin/components/contentClassifier/ContentItemClassifierEditor.vue]: I wonder if it should be v-show instead. v-show works better if we want toggle visibility. V-if on the other is more suitable if the "displaying condition" won't change that often - for instance the v-if from line above.

Comment (rogatty) [reply resources/assets/js/admin/components/contentClassifier/ContentItemClassifierEditor.vue]: There is ~100 flashcards on http://platform.local/app/courses/1/lessons/15/screens/72, I doubt they'll often be all expanded at once.

Comment (bognix) [reply resources/assets/js/admin/components/contentClassifier/ContentItemClassifierEditor.vue]: I see, you don't want to render this editor so many times on page.

File: resources/assets/js/consts/contentClassifier.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const CONTENT_TYPES = {
+	ANNOTATION: 'annotation',
+	FLASHCARD: 'flashcard',
+	QUIZ_QUESTION: 'quizQuestions',
```
Comment (rogatty) [inline resources/assets/js/consts/contentClassifier.js]: `flashcard` but `quizQuestions`?

File: resources/assets/js/components/collections/QuizCollection.vue [hunk 3]
```diff
@@ -97,6 +104,12 @@ export default {
 				feature: features.quiz_questions.value,
 				...payload,
 			});
+		},
+		onTaxonomyTermAttached(payload) {
+			this.attachTerm(payload);
```
Comment (kvas-damian) [inline resources/assets/js/components/collections/QuizCollection.vue]: Is it ok that we call mutation directly form a component?

Comment (bognix) [reply resources/assets/js/components/collections/QuizCollection.vue]: you would like to call it through action?

Comment (kvas-damian) [reply resources/assets/js/components/collections/QuizCollection.vue]: I'm not sure if there is any convention for that, but calling mutation was surprising.

File: resources/assets/js/components/course/screens/flashcards/FlashcardItem.vue [hunk 4]
```diff
@@ -302,7 +308,33 @@ export default {
 		},
 		toggleNoteEditor() {
 			this.isNoteEditorOpen = !this.isNoteEditorOpen;
-		}
+		},
+		onTaxonomyTermAttached(term) {
+			if (!this.flashcard.taxonomyTerms.find(({id}) => id === term.id)) {
```
Comment (bognix) [inline resources/assets/js/components/course/screens/flashcards/FlashcardItem.vue]: I don't like the fact that we are duplicating the same or almost the same code in:
- Annotations
- Flashcards
- Quiz Questions
- Slides

I wonder if solution like:
- scoped slot
- shared vuex store
don't know. I'm thinking out loud.

File: resources/assets/js/components/course/screens/slideshow/SlideClassifierEditor.vue [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+		},
+		onTermDetached(term) {
+			const termIndex = this.slideWithTerms.taxonomyTerms.findIndex(({id}) => id === term.id);
+			this.slideWithTerms.taxonomyTerms.splice(termIndex, 1);
+		}
+	},
+	watch: {
+		currentSlideId(currentSlideId) {
+			if (this.canAccess) {
+				this.loadTerms(currentSlideId);
```
Comment (bognix) [inline resources/assets/js/components/course/screens/slideshow/SlideClassifierEditor.vue]: debounce please.

File: resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue [hunk 6]
```diff
@@ -4,7 +4,7 @@
 		'is-loading': isLoading,
 	}">
 		<h4 class="title is-4 margin bottom">Przypisane pojęcia</h4>
-		<div v-if="this.filteredContent.length > 0">
+		<div v-if="this.items.length > 0">
```
Comment (bognix) [inline resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue]: no need to use `this` inside template.

File: resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+			padding: $margin-tiny
+
+</style>
+
+<script>
+import {mapGetters} from 'vuex';
+import WnlContentClassifierEditor from 'js/components/global/contentClassifier/ContentClassifierEditor';
+
+const CONTENT_TYPE_NAMES = {
+	flashcards: 'Pytanie otwarte',
```
Comment (bognix) [inline resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue]: keys in this object could be fetched from CONTENT_TYPES object

File: resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue [hunk 8]
```diff
@@ -117,6 +118,7 @@ import {ALERT_TYPES} from 'js/consts/alert';
 import WnlSelect from 'js/admin/components/forms/Select';
 import WnlTaxonomyTermAutocomplete from 'js/admin/components/taxonomies/TaxonomyTermAutocomplete';
 import WnlTaxonomyTermWithAncestors from 'js/admin/components/taxonomies/TaxonomyTermWithAncestors';
```
Comment (rogatty) [inline resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue]: Do we want to move these to global components too?

File: resources/assets/js/components/questions/ActiveQuestion.vue [hunk 9]
```diff
@@ -28,6 +28,8 @@
 			@selectAnswer="selectAnswer"
 			@answerDoubleclick="onAnswerDoubleclick"
 			@userEvent="proxyUserEvent"
+			@taxonomyTermAttached="(payload) => $emit('taxonomyTermAttached', payload)"
```
Comment (bognix) [inline resources/assets/js/components/questions/ActiveQuestion.vue]: check if notation `$emit(taxonomyTermAttached, $event)` will work.

File: resources/assets/js/store/modules/questions.js [hunk 10]
```diff
@@ -246,6 +248,18 @@ const mutations = {
 	},
 	[types.QUESTIONS_SET_TOKEN] (state) {
 		state.token = uuidv1();
+	},
+	[types.QUESTIONS_ATTACH_TERM] (state, {question, term}) {
+		const hasTerm = question.taxonomyTerms.find(questionTerm => questionTerm.id === term.id) > -1;
```
Comment (bognix) [inline resources/assets/js/store/modules/questions.js]: findIndex, not find.

File: resources/assets/js/store/modules/questions.js [hunk 11]
```diff
@@ -387,14 +411,24 @@ const actions = {
...
 			const comments = _.get(included, 'comments');
 			comments && dispatch('comments/setComments', comments, {root:true});
 
-			commit(types.QUESTIONS_SET_TEST, {answers, questions, slides});
+			const parsedQuestions = questions.map(question => ({
+				...question,
+				type: CONTENT_TYPES.QUIZ_QUESTION,
+				taxonomyTerms: parseTaxonomyTermsFromIncludes(question.taxonomy_terms, included),
+			}));
+			delete included.tags;
```
Comment (bognix) [inline resources/assets/js/store/modules/questions.js]: just something for discussion, instead of using delete keyword which isn't really performant, maybe it's better to use destruction to get items we are interested in and using rest operator collect remaining includes.

File: resources/assets/js/store/modules/quiz.js [hunk 12]
```diff
@@ -236,6 +238,18 @@ const mutations = {
 	},
 	[types.QUIZ_SET_PAGINATION] (state, pagination) {
 		set(state, 'pagination', pagination);
+	},
+	[types.QUIZ_QUESTION_ATTACH_TERM] (state, {question, term}) {
+		const hasTerm = question.taxonomyTerms.find(questionTerm => questionTerm.id === term.id) > -1;
```
Comment (bognix) [inline resources/assets/js/store/modules/quiz.js]: `findIndex`

File: resources/assets/js/components/collections/QuizCollection.vue (hunk 3 above)
Comment (bognix) [reply resources/assets/js/components/collections/QuizCollection.vue]: well, we do that in flashcards already. It's nothing new.

File: resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue (hunk 8 above)
Comment (bognix) [reply resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue]: you are right 👍

File: resources/assets/js/components/collections/QuizCollection.vue [hunk 13]
```diff
@@ -15,6 +15,8 @@
 				@verify="trackAndResolve"
 				@selectAnswer="onSelectAnswer"
 				@userEvent="onUserEvent"
+				@taxonomyTermAttached="onTaxonomyTermAttached"
+				@taxonomyTermDetached="onTaxonomyTermDetached"
```
Comment (bognix) [inline resources/assets/js/components/collections/QuizCollection.vue]: this thing can be removed.

File: resources/assets/js/components/collections/QuizCollection.vue [hunk 14]
```diff
@@ -38,13 +40,14 @@
 </style>
 
 <script>
-import {mapActions, mapGetters, mapState} from 'vuex';
+import {mapActions, mapGetters, mapState, mapMutations} from 'vuex';
```
Comment (bognix) [inline resources/assets/js/components/collections/QuizCollection.vue]: this change can be removed.

File: resources/assets/js/components/collections/QuizCollection.vue [hunk 15]
```diff
@@ -38,13 +40,14 @@
...
 
 import QuizWidget from 'js/components/quiz/QuizWidget';
 import Pagination from 'js/components/global/Pagination';
 import emits_events from 'js/mixins/emits-events';
 import features from 'js/consts/events_map/features.json';
 import feature_components from 'js/consts/events_map/feature_components.json';
+import { QUIZ_QUESTION_ATTACH_TERM, QUIZ_QUESTION_DETACH_TERM } from 'js/store/mutations-types';
```
Comment (bognix) [inline resources/assets/js/components/collections/QuizCollection.vue]: this change can be removed.

File: resources/assets/js/components/collections/QuizCollection.vue [hunk 16]
```diff
@@ -69,6 +72,10 @@ export default {
 	},
 	methods: {
 		...mapActions('quiz', ['shuffleAnswers', 'changeQuestion', 'resolveQuestion', 'commitSelectAnswer']),
+		...mapMutations('quiz', {
```
Comment (bognix) [inline resources/assets/js/components/collections/QuizCollection.vue]: this change can be removed

File: resources/assets/js/components/collections/QuizCollection.vue [hunk 17]
```diff
@@ -97,6 +104,12 @@ export default {
 				feature: features.quiz_questions.value,
 				...payload,
 			});
+		},
+		onTaxonomyTermAttached(payload) {
```
Comment (bognix) [inline resources/assets/js/components/collections/QuizCollection.vue]: this change can be removed

File: resources/assets/js/components/course/screens/slideshow/SlideClassifierEditor.vue [hunk 18]
```diff
@@ -0,0 +1,49 @@
+<template>
+		<wnl-content-item-classifier-editor
+			:content-item-id="currentSlideId"
+			:content-item-type="CONTENT_TYPES.SLIDE"
+		/>
+</template>
+
+<style lang="sass" rel="stylesheet/sass" scoped>
```
Comment (bognix) [inline resources/assets/js/components/course/screens/slideshow/SlideClassifierEditor.vue]: this change can be removed

File: resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+		},
+		contentItem() {
+			return this.getContentItem({contentItemType: this.contentItemType, contentItemId: this.contentItemId}) || {};
+		},
+		hasContentItem() {
+			return this.contentItem.id;
+		},
+	},
+	methods: {
+		...mapActions('contentClassifier', ['fetchTaxonomyTerms']),
```
Comment (bognix) [inline resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue]: this change can be removed

File: resources/assets/js/components/questions/QuestionsList.vue [hunk 20]
```diff
@@ -276,7 +276,9 @@ export default {
 			'setPage',
 			'fetchActiveFilters',
 		]),
-		...mapMutations('questions', {setToken}),
+		...mapMutations('questions', {
+			setToken,
+		}),
```
Comment (bognix) [inline resources/assets/js/components/questions/QuestionsList.vue]: revert this change.

File: resources/assets/js/components/questions/QuestionsTest.vue [hunk 21]
```diff
@@ -84,6 +84,8 @@
 			:plainList="true"
 			@selectAnswer="selectAnswer"
 			@userEvent="onUserEvent"
+			@taxonomyTermAttached="$emit('taxonomyTermAttached', $event)"
```
Comment (bognix) [inline resources/assets/js/components/questions/QuestionsTest.vue]: this change can be removed

File: resources/assets/js/components/questions/QuestionsTestBuilder.vue [hunk 22]
```diff
@@ -12,6 +12,8 @@
 			@userEvent="proxyUserEvent"
 			@testStart="onTestStart"
 			@updateTime="(payload) => $emit('updateTime', payload)"
+			@taxonomyTermAttached="$emit('taxonomyTermAttached', $event)"
+			@taxonomyTermDetached="$emit('taxonomyTermDetached', $event)"
```
Comment (bognix) [inline resources/assets/js/components/questions/QuestionsTestBuilder.vue]: this change can be removed

File: resources/assets/js/components/quiz/Quiz.vue [hunk 23]
```diff
@@ -5,7 +5,9 @@
 				Gratulacje! <wnl-emoji name="tada"></wnl-emoji>
 			</p>
 			<p class="big">Wszystkie pytania rozwiązane poprawnie! Możesz teraz sprawdzić poprawne odpowiedzi, oraz procentowy rozkład wyborów innych uczestników.</p>
-			<wnl-quiz-summary @userEvent="onUserEvent"/>
+			<wnl-quiz-summary
```
Comment (bognix) [inline resources/assets/js/components/quiz/Quiz.vue]: this change can be reverted.

File: resources/assets/js/components/quiz/SingleQuestion.vue [hunk 24]
```diff
@@ -66,11 +68,12 @@
 </style>
 
 <script>
-import { mapActions, mapGetters } from 'vuex';
+import { mapActions, mapGetters, mapMutations } from 'vuex';
```
Comment (bognix) [inline resources/assets/js/components/quiz/SingleQuestion.vue]: this change can be removed

File: resources/assets/js/components/quiz/SingleQuestion.vue [hunk 25]
```diff
@@ -66,11 +68,12 @@
...
 
 import WnlQuestionsNavigation from 'js/components/questions/QuestionsNavigation';
 import WnlQuizWidget from 'js/components/quiz/QuizWidget';
 import emits_events from 'js/mixins/emits-events';
+import { QUIZ_QUESTION_ATTACH_TERM, QUIZ_QUESTION_DETACH_TERM } from 'js/store/mutations-types';
```
Comment (bognix) [inline resources/assets/js/components/quiz/SingleQuestion.vue]: this change can be removed

File: resources/assets/js/components/quiz/SingleQuestion.vue [hunk 26]
```diff
@@ -99,6 +102,10 @@ export default {
 	},
 	methods: {
 		...mapActions('quiz', ['destroyQuiz', 'fetchSingleQuestion', 'commitSelectAnswer', 'resolveQuestion']),
+		...mapMutations('quiz', {
```
Comment (bognix) [inline resources/assets/js/components/quiz/SingleQuestion.vue]: this change can be removed

File: resources/assets/js/components/quiz/SingleQuestion.vue [hunk 27]
```diff
@@ -115,6 +122,12 @@ export default {
 					this.hasError = true;
 				});
 		},
+		onTaxonomyTermAttached(payload) {
```
Comment (bognix) [inline resources/assets/js/components/quiz/SingleQuestion.vue]: this change can be removed

File: resources/assets/js/store/modules/questions.js [hunk 28]
```diff
@@ -17,6 +17,8 @@ import {
 	reactionsGetters,
 	reactionsMutations
 } from 'js/store/modules/reactions';
+import {parseTaxonomyTermsFromIncludes} from 'js/utils/contentClassifier';
```
Comment (bognix) [inline resources/assets/js/store/modules/questions.js]: changes in this file can be reverted.

File: resources/assets/js/store/modules/quiz.js [hunk 29]
```diff
@@ -6,6 +6,8 @@ import {commentsGetters, commentsMutations, commentsActions, commentsState} from
 import {reactionsGetters, reactionsMutations, reactionsActions} from 'js/store/modules/reactions';
 import * as types from 'js/store/mutations-types';
 import quizStore from 'js/services/quizStore';
+import {parseTaxonomyTermsFromIncludes} from 'js/utils/contentClassifier';
+import {CONTENT_TYPES} from 'js/consts/contentClassifier';
```
Comment (bognix) [inline resources/assets/js/store/modules/quiz.js]: changes in this file can be reverted.

File: resources/assets/js/store/mutations-types.js [hunk 30]
```diff
@@ -118,6 +118,8 @@ export const QUIZ_RESTORE_STATE     = 'QUIZ_RESTORE_STATE';
 export const QUIZ_TOGGLE_PROCESSING = 'QUIZ_TOGGLE_PROCESSING';
 export const QUIZ_SET_STATS         = 'QUIZ_SET_STATS';
 export const QUIZ_RESET_PROGRESS    = 'QUIZ_RESET_PROGRESS';
+export const QUIZ_QUESTION_ATTACH_TERM = 'QUIZ_QUESTION_ATTACH_TERM';
+export const QUIZ_QUESTION_DETACH_TERM = 'QUIZ_QUESTION_DETACH_TERM';
```
Comment (bognix) [inline resources/assets/js/store/mutations-types.js]: this change can be removed

File: resources/assets/js/store/mutations-types.js [hunk 31]
```diff
@@ -171,6 +173,9 @@ export const QUESTIONS_SET_TEST                 = 'QUESTIONS_SET_TEST';
 export const QUESTIONS_SET_WITH_ANSWERS         = 'QUESTIONS_SET_WITH_ANSWERS';
 export const QUESTIONS_UPDATE                   = 'QUESTIONS_UPDATE';
 export const QUESTIONS_SET_TOKEN                = 'QUESTIONS_SET_TOKEN';
+export const QUESTIONS_ATTACH_TERM              = 'QUESTIONS_ATTACH_TERM';
+export const QUESTIONS_DETACH_TERM              = 'QUESTIONS_DETACH_TERM';
```
Comment (bognix) [inline resources/assets/js/store/mutations-types.js]: this change can be removed

File: resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue [hunk 32]
```diff
@@ -99,6 +99,8 @@
 
 <script>
 import { getColourForStr } from 'js/utils/colors.js';
+import {parseTaxonomyTermsFromIncludes} from 'js/utils/contentClassifier';
+import {CONTENT_TYPES} from 'js/consts/contentClassifier';
```
Comment (kvas-damian) [inline resources/assets/js/admin/components/slides/annotations/AnnotationsList.vue]: Not used

File: resources/assets/js/components/course/screens/slideshow/SlideClassifierEditor.vue [hunk 33]
```diff
@@ -0,0 +1,45 @@
+<template>
+		<wnl-content-item-classifier-editor
```
Comment (kvas-damian) [inline resources/assets/js/components/course/screens/slideshow/SlideClassifierEditor.vue]: indentation

File: resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue [hunk 34]
```diff
@@ -250,6 +255,8 @@ export default {
 		}
 	},
 	async mounted() {
+		if (this.taxonomiesSet) return;
```
Comment (kvas-damian) [inline resources/assets/js/components/global/contentClassifier/ContentClassifierEditor.vue]: maybe we could move this condition to store?

File: resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue [hunk 35]
```diff
@@ -0,0 +1,129 @@
...
+			required: true
+		}
+	},
+	computed: {
+		...mapGetters('contentClassifier', ['getContentItem', 'canAccess']),
+		hasTaxonomyTerms() {
+			return this.contentItem.taxonomyTerms && this.contentItem.taxonomyTerms.length > 0;
+		},
+		contentItem() {
+			return this.getContentItem({contentItemType: this.contentItemType, contentItemId: this.contentItemId}) || {};
```
Comment (kvas-damian) [inline resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue]: `null` as default value?

File: resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue [hunk 36]
```diff
@@ -0,0 +1,129 @@
+<template>
+	<div v-if="canAccess && hasContentItem" class="content-item-classifier">
```
Comment (kvas-damian) [inline resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue]: Loading state?

File: resources/assets/js/store/modules/contentClassifier.js [hunk 37]
```diff
@@ -0,0 +1,84 @@
...
+
+		set(state, contentType, {
+			...state[contentType],
+			...reducedState
+		});
+	}
+};
+
+const actions = {
+	async fetchTaxonomyTerms({commit, getters}, {contentType, contentIds}) {
```
Comment (kvas-damian) [inline resources/assets/js/store/modules/contentClassifier.js]: error handling

File: resources/assets/js/store/modules/contentClassifier.js [hunk 38]
```diff
@@ -0,0 +1,84 @@
...
+	canAccess: (state, getters, rootState, rootGetters) => rootGetters.isAdmin || rootGetters.isModerator
+};
+
+const mutations = {
+	[mutationsTypes.CONTENT_CLASSIFIER_ATTACH_TERM](state, {term, contentItem}) {
+		if (contentItem.taxonomyTerms.findIndex(taxonomyTerm => taxonomyTerm.id === term.id) < 0) {
+			contentItem.taxonomyTerms.push(term);
+		}
+	},
+	[mutationsTypes.CONTENT_CLASSIFIER_DETACH_TERM](state, {term, contentItem}) {
```
Comment (kvas-damian) [inline resources/assets/js/store/modules/contentClassifier.js]: Is it a store mutation?
Maybe we should use an util method like `parseTaxonomyTermsFromIncludes`?

File: resources/assets/js/store/modules/contentClassifier.js [hunk 39]
```diff
@@ -0,0 +1,84 @@
...
+		}
+	},
+	[mutationsTypes.CONTENT_CLASSIFIER_SET_CONTENT](state, {contentItems, contentType}) {
+		const reducedState = contentItems.reduce((items, item) => {
+			items[item.id] = item;
+			return items;
+		}, {});
+
+		set(state, contentType, {
+			...state[contentType],
```
Comment (kvas-damian) [inline resources/assets/js/store/modules/contentClassifier.js]: we never remove items from the store, also we always fetch fresh data from API. Is ok? Or do we want to save the browser from handling really big observables list?

File: resources/assets/js/store/modules/contentClassifier.js (hunk 37 above)
Comment (kvas-damian) [reply resources/assets/js/store/modules/contentClassifier.js]: while this functionality isn't critical in day-to-day work I'd suggest using a small error message inside `ContentClassifierItemEditor` instead of dismissible alert. This way we keep the user informed what is going on and in the time we don't disturb him too much when he is focused on other things.
What do you think?

File: resources/assets/js/store/modules/contentClassifier.js (hunk 38 above)
Comment (bognix) [reply resources/assets/js/store/modules/contentClassifier.js]: it's mutating the store. The contentItem is coming from store.

File: resources/assets/js/store/modules/contentClassifier.js (hunk 39 above)
Comment (bognix) [reply resources/assets/js/store/modules/contentClassifier.js]: Yes, that's true. Don't know if we want to do pre-optimization. We use this approach in other places as well. I just followed the pattern. Maybe it's worth to create more reusable approach to don't download things we already have in store.

File: resources/assets/js/store/modules/contentClassifier.js (hunk 37 above)
Comment (bognix) [reply resources/assets/js/store/modules/contentClassifier.js]: makes sense but this thing is not called in ContentClassifierItemEditor.

File: resources/assets/js/store/modules/contentClassifier.js (hunk 39 above)
Comment (kvas-damian) [reply resources/assets/js/store/modules/contentClassifier.js]: Ok, let's leave it for now

File: resources/assets/js/store/modules/contentClassifier.js (hunk 37 above)
Comment (kvas-damian) [reply resources/assets/js/store/modules/contentClassifier.js]: Done

File: resources/assets/js/store/modules/contentClassifier.js (hunk 38 above)
Comment (kvas-damian) [reply resources/assets/js/store/modules/contentClassifier.js]: Any ideas on how to make that obvious?

File: resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue [hunk 40]
```diff
@@ -0,0 +1,139 @@
...
+			</div>
+			<div v-else class="clickable content-item-classifier__tag-names" @click="expanded=true">
+				<span class="content-item-classifier__tag-icon icon is-small"><i class="fa fa-tags"></i></span>
+				<span v-if="hasTaxonomyTerms">{{contentItem.taxonomyTerms.map(term => term.tag.name).join(', ')}}</span>
+				<span v-else>brak</span>
+			</div>
+		</template>
+		<div v-else-if="isError">
+			<span class="content-item-classifier__tag-icon icon is-small"><i class="fa fa-exclamation"></i></span>
+			Ups, nie udalo się załadować przypisanych pojęć. Odśwież strone, aby spróbować ponownie.
```
Comment (bognix) [inline resources/assets/js/components/global/contentClassifier/ContentItemClassifierEditor.vue]: odśwież stron**ę**

File: resources/assets/js/components/quiz/QuizQuestion.vue [hunk 41]
```diff
@@ -200,8 +200,12 @@
 				display: inline-block
 				padding-right: $margin-tiny * 2
 
+	.wnl-quiz-question-container
+		margin-bottom: $margin-big
```
Comment (bognix) [inline resources/assets/js/components/quiz/QuizQuestion.vue]: Did you check how it look for regular users that don't have editor visible?  The margin is smaller and questions on the page may end up more cluttered

File: resources/assets/js/store/modules/contentClassifier.js (hunk 38 above)
Comment (bognix) [reply resources/assets/js/store/modules/contentClassifier.js]: We discussed this offline. Are you going to implement this or should I?

File: resources/assets/js/store/modules/contentClassifier.js [hunk 42]
```diff
@@ -0,0 +1,116 @@
...
+			...reducedState
+		});
+	},
+	[mutationsTypes.CONTENT_CLASSIFIER_SET_ERROR](state, {contentItemIds, contentType}) {
+		contentItemIds.forEach(id => {
+			set(state[contentType][id], 'state', REQUEST_STATES.ERROR);
```
Comment (bognix) [inline resources/assets/js/store/modules/contentClassifier.js]: Did you check how performant this is to do mutations in a for loop.

Because if it works fine, I don't see a reason why we wouldn't like to do this in mutations above:
- SET_CONTENT
- SET_LOADING

File: resources/assets/js/components/quiz/QuizQuestion.vue (hunk 41 above)
Comment (kvas-damian) [reply resources/assets/js/components/quiz/QuizQuestion.vue]: Looks good

File: resources/assets/js/store/modules/contentClassifier.js (hunk 42 above)
Comment (kvas-damian) [reply resources/assets/js/store/modules/contentClassifier.js]: It's not, I changed that
